Code/Tools/FBuild/FBuildCore/BFF/Tokenizer/BFFTokenizer.cpp

```diff
@@ -572,52 +572,108 @@ bool BFFTokenizer::HandleDirective_If( const BFFFile & file, const char * & pos,
     ASSERT( argsIter->IsKeyword( "if" ) );
     argsIter++;
 
-    // Check for negation operator
-    bool negate = false;
-    if ( argsIter->IsOperator( "!" ) )
-    {
-        negate = true;
-        argsIter++; // consume negation operator
-    }
+    enum { IF_NONE = 0, IF_AND = 1, IF_OR = 2, IF_NEGATE = 4 };
```

Comment:
You might like to consider using an enum class here:
```C++
enum OperatorType : class uin32_t { IF_NONE .... }
```
This prevents accidental misuse.

Reply:
Hi,

I tried using enum class as you suggested but I stumbled in a few problems. Most noticeable is that I now cannot use the enum members in boolean arithmetic. For example, I do something like

```
ifOperator |= IF_NEGATE;
```

Adding the enum class as you suggest I either have to declare `ifOperator` as an `OperatorType` type or I have to use casts everywhere: `(uint32_t)OperatorType::IF_NEGATE`. And even that doesn't solve the boolean problem stated above. Reading some questions online I saw that you can overcome this by writing specific functions that handle this.

My knowledge of C++ is sketchy at best, so I might be missing something here. But this seems unnecessarily overkill and makes the code more unreadable than it is originally.

Again, any help is appreciated because I'm stuck with this for now and will revert back to normal enums for the reworking of the code (operator precedence).

Reply:
I think if you're using it as a bit field, it's reasonable to go back to what you had. Apologies for the confusion.